doc/microsite/src/main/tut/distage/distage-framework-docker.md

```diff
@@ -19,3 +20,487 @@ libraryDependencies += "io.7mind.izumi" %% "distage-framework-docker" % "$izumi.
+
+To use this container a module that declares this component is required:
+
+##### `make`
+
+```scala mdoc:silent
+class PostgresDockerModule[F[_]: TagK] extends ModuleDef {
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[F]
+  }
+}
+```
+
+##### `modifyConfig`
+
+The @scaladoc[`DockerProviderExtensions`](izumi.distage.docker.DockerContainer.DockerProviderExtensions)
+provides additional APIs for modiying the container definition.
+
+Use `modifyConfig` to modify the configuration of a container. The modifier is instantiated to a
+`ProviderMagnet` which will summon any additional dependencies.
+
+For example, To change the user of the postgres container:
+
+```scala mdoc:silent
+class PostgresRunAsAdminModule[F[_]: TagK] extends ModuleDef {
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[F].modifyConfig { () => (old: PostgresDocker.Config) =>
+      old.copy(user = Option("admin"))
+    }
+  }
+}
+```
+
+Suppose the `HostHtmlRoot` is a component provided by the application modules. This path
+can be added to the postgres containers mounts by adding this to the additional dependencies of
+the provider magnet:
+
+```scala mdoc:silent
+case class HostPostgresData(path: String)
+
+class PostgresWithMountsDockerModule[F[_]: TagK] extends ModuleDef {
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[F].modifyConfig {
+      (hostPostgresData: HostPostgresData) =>
+        (old: PostgresDocker.Config) =>
+          val dataMount = Docker.Mount(hostPostgresData.path, "/var/lib/postgresql/data")
+          old.copy(mounts = old.mounts :+ dataMount)
+    }
+  }
+}
+```
+
+##### `dependOnDocker`
+
+`dependOnDocker` adds a dependency on a given docker container. distage ensures the requested
+container is available before the dependent is provided.
+
+Suppose the system under test is a sync from postgres to elasticsearch. One option is to use
+`dependOnDocker` to declare the elasticsearch container depends on the postgres container:
+
+```scala mdoc:silent
+
+object ElasticSearchDocker extends ContainerDef {
+  val ports = Seq(9200, 9300)
+
+  override def config: Config = {
+    Config(
+      image = "docker.elastic.co/elasticsearch/elasticsearch:7.7.0",
+      ports = ports.map(Docker.DockerPort.TCP(_)),
+      env = Map("discovery.type" -> "single-node")
+    )
+  }
+}
+
+class ElasticSearchPlusPostgresModule[F[_]: TagK] extends ModuleDef {
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[F]
+  }
+
+  make[ElasticSearchDocker.Container].fromResource {
+    ElasticSearchDocker.make[F].dependOnDocker(PostgresDocker)
+  }
+}
+```
+
+Another example of dependencies between containers is in "Docker Container Networks" below.
+
+#### Include `DockerSupportModule`
+
+Include the `izumi.distage.docker.modules.DockerSupportModule` module in the application
+modules. This module contains required component declarations and loads the `docker-reference.conf`.
```

Comment:
It will not load the `docker-reference.conf`, it will only read the `docker` section in application config if it exists.